src/clients/zendesk.ts

```diff
@@ -1,14 +1,63 @@
-import zendesk, {Client, ClientOptions} from 'node-zendesk';
+import zendesk, {ClientOptions} from 'node-zendesk';
+
+import {AppContext} from 'mattermost-redux/types/apps';
 
 import {Env} from '../utils';
+import {oauthStore} from '../store';
+
+interface Tickets {
+    create(ticket: any): any;
+    exportAudit(id: number): any;
+}
+
+interface TicketForms {
+    list(): any;
+}
+
+interface TicketFields {
+    list(): any;
+}
+
+interface Triggers {
+    update(id: number): any;
+    delete(id: number): any;
+    create(trigger: any): any;
+    search(query: string): any;
+}
+
+interface OauthTokens {
+    current(): any;
+    revoke(id: number): any;
+}
+
+interface Users {
+    show(id: number): any;
+}
+
+// expose on the methods that will be used by the app
+export interface ZDClient {
+    tickets: Tickets;
+    ticketforms: TicketForms;
+    ticketfields: TicketFields;
+    triggers: Triggers;
+    oauthtokens: OauthTokens;
+    users: Users;
+}
+
+export const newZDClient = (context: AppContext): ZDClient => {
+    // get active mattermost user ID
+    const mmUserID = context.acting_user_id || '';
```

Comment:
Is there a case where `context.acting_user_id` is undefined?

Reply:
put into catch-all NITs ticket https://mattermost.atlassian.net/browse/MM-33787